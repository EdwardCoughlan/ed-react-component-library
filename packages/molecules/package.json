{
  "name": "@edwardcoughlan/molecules",
  "version": "0.0.1",
  "dependencies": {
    "@edwardcoughlan/atoms": "*"
  },
  "peerDependencies": {
    "react": "^18.1.0",
    "react-dom": "^18.1.0",
    "classnames": "^2.3.1"
  }
}
