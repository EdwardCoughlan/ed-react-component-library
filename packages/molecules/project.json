{
  "sourceRoot": "packages/molecules/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "build": {
      "executor": "@nrwl/web:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/packages/molecules",
        "tsConfig": "packages/molecules/tsconfig.lib.json",
        "project": "packages/molecules/package.json",
        "entryFile": "packages/molecules/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": "@nrwl/react/plugins/bundle-rollup",
        "assets": [
          {
            "glob": "packages/molecules/README.md",
            "input": ".",
            "output": "."
          }
        ]
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["packages/molecules/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/packages/molecules"],
      "options": {
        "jestConfig": "packages/molecules/jest.config.ts",
        "passWithNoTests": true
      }
    }
  }
}
