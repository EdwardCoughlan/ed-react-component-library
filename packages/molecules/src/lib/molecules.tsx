import './molecules.module.scss';
import { Atoms } from '@edwardcoughlan/atoms';

/* eslint-disable-next-line */
export interface MoleculesProps {}

export function Molecules(): JSX.Element {
  return <Atoms welcomeMessage="Hi from molecules" />;
}
