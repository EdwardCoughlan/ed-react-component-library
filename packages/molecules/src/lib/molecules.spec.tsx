import { render } from '@testing-library/react';

import { Molecules } from './molecules';

describe('Molecules', () => {
  it('should render successfully', () => {
    const { baseElement, container } = render(<Molecules />);
    expect(baseElement).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
