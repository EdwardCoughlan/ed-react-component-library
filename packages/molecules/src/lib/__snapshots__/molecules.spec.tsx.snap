// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules should render successfully 1`] = `
<div>
  <div
    class="background"
  >
    <h1>
      Hi from molecules
    </h1>
  </div>
</div>
`;
